Makefile

```diff
@@ -3,11 +3,12 @@ include config.mk
 XAPIDOC=_build/install/default/xapi/doc
 JOBS = $(shell getconf _NPROCESSORS_ONLN)
 PROFILE=release
+XAPI_VERSION ?= $(shell git describe --always --dirty)
```

Comment:
This can fail when there is no git context. Would it be better to account for this?
```
API_VERSION ?= $(shell git describe --always --dirty || echo "000000")
```